contracts/zero-ex/contracts/src/features/OtcOrdersFeature.sol

```diff
@@ -0,0 +1,443 @@
+            order,
+            taker,
+            takerTokenFillAmount,
+            wethOptions
+        );
+
+        emit OtcOrderFilled(
+            orderInfo.orderHash,
+            order.maker,
+            taker,
+            address(order.makerToken),
+            address(order.takerToken),
+            takerTokenFilledAmount,
+            makerTokenFilledAmount
+        );
+    }
+
+    /// @dev Settle the trade between an OTC order's maker and taker.
+    /// @param order The OTC order.
+    /// @param takerTokenFillAmount Maximum taker token amount to fill this
+    ///        order with.
+    /// @return takerTokenFilledAmount How much taker token was filled.
+    /// @return makerTokenFilledAmount How much maker token was filled.
+    function _settleOtcOrder(
+        LibNativeOrder.OtcOrder memory order,
+        address taker,
+        uint128 takerTokenFillAmount,
+        WethOptions wethOptions
+    )
+        private
+        returns (uint128 takerTokenFilledAmount, uint128 makerTokenFilledAmount)
+    {
+        // Update tx origin nonce for the order.
+        LibOtcOrdersStorage.getStorage().txOriginNonces
+            [order.txOrigin][order.txOriginNonceBucket] = order.txOriginNonce;
+
+        // Clamp the taker token fill amount to the fillable amount.
+        takerTokenFilledAmount = LibSafeMathV06.min128(
+            takerTokenFillAmount,
+            order.takerAmount
+        );
+        // Compute the maker token amount.
+        // This should never overflow because the values are all clamped to
+        // (2^128-1).
+        makerTokenFilledAmount = uint128(LibMathV06.getPartialAmountFloor(
+            uint256(takerTokenFilledAmount),
+            uint256(order.takerAmount),
+            uint256(order.makerAmount)
+        ));
+
+        if (wethOptions == WethOptions.WrapEth) {
+            require(
+                order.takerToken == WETH,
+                "OtcOrdersFeature/INVALID_WRAP_ETH"
+            );
+            // Wrap ETH
+            WETH.deposit{value: takerTokenFilledAmount}();
+            // Transfer WETH to maker
+            WETH.transfer(order.maker, takerTokenFilledAmount);
+            if (takerTokenFilledAmount < msg.value) {
+                // Refund unused ETH
+                _transferEth(
+                    msg.sender,
+                    msg.value - uint256(takerTokenFilledAmount)
+                );
+            }
+        } else {
+            // Transfer taker -> maker
+            _transferERC20Tokens(
+                order.takerToken,
+                taker,
+                order.maker,
+                takerTokenFilledAmount
+            );
+        }
+
+        if (wethOptions == WethOptions.UnwrapWeth) {
+            require(
+                order.makerToken == WETH,
+                "OtcOrdersFeature/INVALID_UNWRAP_WETH"
+            );
+            // Transfer maker tokens in
+            _transferERC20Tokens(
```

Comment:
Isn't this going to be more expensive? because we need to first transfer the tokens to this contract, then unwrap, then send?

Reply:
more expensive than what? yes it's more expensive than thus using WETH, but sometimes people want to receive ETH. This is the cheapest way to do that